llvm/lib/CodeGen/AsmPrinter/DwarfFile.cpp

```diff
@@ -94,7 +125,43 @@ unsigned DwarfFile::computeSizeAndOffset(DIE &Die, unsigned Offset) {
                                       Offset);
 }
 
-void DwarfFile::emitAbbrevs(MCSection *Section) { Abbrevs.Emit(Asm, Section); }
+void DwarfFile::emitAbbrevs(MCSection *Section) {
+  // If the Abbreviation vector only has one Abbreviation contribution and the
+  // contribution has non-zero Abbreviations in use, emit it without emitting
+  // any labels. This is done because there are some targets does not support
+  // labels inside debug sections.
+  if (Abbrevs.size() == 1 && !Abbrevs[0]->isAbbreviationListEmpty()) {
+    Asm->OutStreamer->SwitchSection(Section);
+    Abbrevs[0]->Emit(Asm);
+    return;
+  }
+  for (unsigned i = 0; i < Abbrevs.size(); i++) {
+    // If the Abbreviation contribution has no Abbreviations in use, do not
+    // switch sections or emit labels.
+    if (!Abbrevs[i]->isAbbreviationListEmpty()) {
+      Asm->OutStreamer->SwitchSection(Section);
+      MCSymbol *AbbrevStartSym = getOrCreateAbbrevSectionStart();
+      // Do not emit the same label again, AbbrevStartSym should be the same for
+      // all Abbreviation contributions, it denotes the start of the
+      // Abbreviation Section.
+      if (!AbbrevStartEmitted) {
+        AbbrevStartEmitted = true;
+        Asm->OutStreamer->emitLabel(AbbrevStartSym);
+      }
+      if (CUs.size() && i < CUs.size()) {
```

Comment:
Is there any legitimate reason why i would be > than CUs.size() that is not a programming error?
If no, should this be an assert?